CMakeLists.txt

```diff
@@ -1,5 +1,13 @@
 cmake_minimum_required(VERSION 3.2)
 
+# If cbmc generates files inside of the cbmc source directory this can lead to important files being overwritten, so we prevent that here
+if("${CMAKE_BINARY_DIR}" STREQUAL "${CMAKE_SOURCE_DIR}")
+  message(FATAL_ERROR
+    "Do not generate CMake files inside of the cbmc source directory."
```

Comment:
I'm not sure about the wording here: you are talking to the user here, who just invoked `cmake` - they didn't actually (intend to) create any files at all, and might thus be confused by this error message.